File: TUF/Sorting/SelectionSort.ts
function selectionSort(arr: number[]): void {
    for (let i: number = 0; i < arr.length - 1; i++) {
        let mini: number = i;
        for (let j: number = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[mini]) {
                mini = j;
            }
        }

        let temp: number = arr[i];
        arr[i] = arr[mini];
        arr[mini] = temp;
    }
}

let arr: number[] = [13, 46, 24, 52, 20, 9];
console.log(`Array before sorting: ${arr}`);
selectionSort(arr);
console.log(`Array after sorting: ${arr}`);

export { };

File: TUF/Sorting/SelectionSort.java
import java.util.Scanner;

public class SelectionSort {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter the %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println("Array before sorting...");
        printArray(arr);
        System.out.println("Array after sorting...");
        selectionSort(arr);
        printArray(arr);
    }

    private static void selectionSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            int mini = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[mini]) {
                    mini = j;
                }
            }

            int temp = arr[i];
            arr[i] = arr[mini];
            arr[mini] = temp;
        }
    }

    private static void printArray(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
